src/main/java/com/iexec/core/chain/BlockchainConnectionHealthIndicator.java

```diff
@@ -0,0 +1,165 @@
+package com.iexec.core.chain;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@Component
+public class BlockchainConnectionHealthIndicator implements HealthIndicator {
+
+    private final Web3jService web3jService;
+    /**
+     * Interval between 2 requests onto the chain.
+     */
+    private final Duration pollingInterval;
+    /**
+     * Number of consecutive failures before declaring this Scheduler is out-of-service.
+     */
+    private final int outOfServiceThreshold;
+    private final ScheduledExecutorService monitoringExecutor;
+
+    /**
+     * Current number of consecutive failures.
+     */
+    private int consecutiveFailures = 0;
+    private LocalDateTime firstFailure = null;
+    private boolean outOfService = false;
```

Comment:
What do you think to add getters to those fields ?
* Simplify tests
* Would allow us to expose them to the metrics endpoint (would be nice to see the graph of temporary communication loss)

Reply:
I'm unsure about the second point. Do we want the information in the `/metrics` endpoint  while we have it in the `/actuator/health/blockchainConnection` endpoint?
If we don't, I'd prefer to keep it as is.

Reply:
The goal of the metric is to send data to a metric aggregator.
The health endpoint is different and should not be scraped by a monitoring endpoint.
If we want to follow communication loss without out of service, we need to do it through a metric imho.
I'd be happy to have a live discussion on this if you want to.

Reply:
I see your point. Here's the commit: https://github.com/iExecBlockchainComputing/iexec-core/pull/601/commits/c49ae6ee11655cc03fdce957687146867d17e4ae